File: backend/SmartTrip.Domain/Entities/Destination.cs
using System;
using System.Collections.Generic;

namespace SmartTrip.Domain.Entities;

public partial class Destination
{
    public int DestId { get; set; }

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    public string? CoverImageUrl { get; set; }

    public bool? IsHot { get; set; }

    public virtual ICollection<BlogPost> BlogPosts { get; set; } = new List<BlogPost>();

    public virtual ICollection<BusSchedule> BusScheduleFromDests { get; set; } = new List<BusSchedule>();

    public virtual ICollection<BusSchedule> BusScheduleToDests { get; set; } = new List<BusSchedule>();

    public virtual ICollection<Hotel> Hotels { get; set; } = new List<Hotel>();

    public virtual ICollection<Trip> Trips { get; set; } = new List<Trip>();
}

File: mobile/lib/models/destination.dart
class Destination {
  final int id;
  final String name;
  final String description;
  final String coverImageUrl;
  final bool isHot;

  const Destination({
    required this.id,
    required this.name,
    required this.description,
    required this.coverImageUrl,
    required this.isHot,
  });

  factory Destination.fromJson(Map<String, dynamic> json) {
    return Destination(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      coverImageUrl: json['coverImageUrl'] ?? '',
      isHot: json['isHot'] ?? false,
    );
  }
}
